src/main/scala/chisel3/util/Decoupled.scala

```diff
@@ -215,9 +216,12 @@ class Queue[T <: Data](val gen: T,
 
   val io = IO(new QueueIO(genType, entries))
 
-  val ram = Mem(entries, genType)
+  val ram = if (useSyncReadMem) SyncReadMem(entries, genType) else Mem(entries, genType)
+  
   val enq_ptr = Counter(entries)
   val deq_ptr = Counter(entries)
+  val deq_ptr_wire = WireDefault(deq_ptr.value)
```

Comment:
should this be called `deq_ptr_value` since that's what it is?

Reply:
also should this entire wire be an Option since it's only used if you have SyncReadMem?

Reply:
Sure, you're right. The name is changed as per your suggestion and the entire wire is an Option now.